webui/react/src/contexts/AppContexts.tsx

```diff
@@ -39,15 +41,31 @@ const AppContexts: React.FC = () => {
   const [ activeExperimentsResponse, triggerActiveExperimentsRequest ] =
     useRestApi<ExperimentsParams, ExperimentBase[]>(getExperimentSummaries, {});
 
+  const fetchAgents = useCallback(async (): Promise<void> => {
+    try {
+      const agentsResponse = await getAgents({ cancelToken: apiSource.token });
+      setAgents({
+        type: Agents.ActionType.Set,
+        value: {
+          data: agentsResponse,
+          errorCount: agents.errorCount,
+          hasLoaded: true,
+          isLoading: false,
+        },
+      });
+      setOverview({ type: ClusterOverview.ActionType.SetAgents, value: agentsResponse });
+    } catch (e) {
+      handleError({ message: 'Unable to fetch agents.', silent: true, type: ErrorType.Api });
```

Comment:
yeah that's a good point. It looks like we can't have a `try` block without either a `catch` or `finally` block following it. I still want the `try` block to wrap around in case an error is thrown to prevent the page from breaking if it's uncaught. So I just allowed an empty catch block by skipping eslint `no-empty` rule.